composer.json

```diff
@@ -6,24 +6,27 @@
     "config": {
         "sort-packages": true,
         "platform": {
-            "php": "8.0.99"
+            "php": "8.1.99"
         },
         "allow-plugins": {
             "dealerdirect/phpcodesniffer-composer-installer": true
         }
     },
     "require": {
-        "php": "~8.0.0 || ~8.1.0 || ~8.2.0 || ~8.3.0",
+        "php": "~8.1.0 || ~8.2.0 || ~8.3.0 || ~8.4.0",
         "composer-plugin-api": ">=1.1.0 <2.3.0"
     },
     "require-dev": {
+        "amphp/dns": "^2.4.0",
+        "amphp/socket": "^2.3.1",
+        "amphp/byte-stream": "^2.0.0",
         "composer/composer": ">=1.9.0 <2.3.0",
         "laminas/laminas-coding-standard": "~2.5.0",
         "mikey179/vfsstream": "^1.6.11",
-        "phpunit/phpunit": "^9.5.5",
-        "psalm/plugin-phpunit": "^0.18.0",
+        "phpunit/phpunit": "^9.6.2",
```

Comment:
use latest ^9.6.22, also please regenerate composer.lock

and after everything green, better squash the commits :)